lib/conrad/processor_stack.rb

```diff
@@ -0,0 +1,47 @@
+require 'forwardable'
+
+module Conrad
+  class ProcessorStack
+    include Enumerable
+    extend Forwardable
+
+    # The processors used by this Stack
+    attr_reader :processors
+
+    # Delegate #each and #to_a to the processors Array to allow iterating over
+    # the object as if it were a collection of processors.
+    def_delegators :processors, :each, :to_a
+
+    # @param processors [Array<#call>] set of objects all responding to #call
+    #   that will be used to process an event
+    def initialize(processors)
+      check_callability(processors)
```

Comment:
is this something we want here, or should this happen naturally during runtime?

Reply:
I, personally, like it on initialization because it's going to tell you right away if this will work rather than failing sometime in the middle of a request. Throwing a fit at the start seems nicer and helps provide a clearer message of "Hey, this thing you gave me isn't going to work, fix it" rather than later, in the middle of processing going, "This thing didn't work because it lacked a method please help." Fail early, fail hard, fail fast. This is making sure someone gives us the right duck.

Reply:
Sounds good! Now that i think about it, i've done the same thing a few times and it *IS* nice when it yells on localhost boot rather than 3 days into being in prod.